<launch>

	<env name="ROSCONSOLE_CONFIG_FILE" value="$(find ros_caffe)/cfg/custom_rosconsole.conf"/>

	<include file="$(find openni_launch)/launch/openni.launch">
		<arg name="load_driver" value="true" />
	</include>

	<node pkg="ros_caffe" type="ros_caffe_node" name="ros_caffe" output="screen">
		<param name="model_path"    value="$(find ros_caffe)/data/deploy.prototxt"/>
		<param name="weights_path"  value="$(find ros_caffe)/data/bvlc_reference_caffenet.caffemodel"/>
		<param name="mean_file"     value="$(find ros_caffe)/data/imagenet_mean.binaryproto"/>
		<param name="label_file"    value="$(find ros_caffe)/data/synset_words.txt"/>
		<param name="image_path"    value="$(find ros_caffe)/data/cat.jpg"/>
		<param name="test_image"    value="true"/>

		<remap from="~image"        to="/camera/rgb/image_raw"/>
		<remap from="~camera_info"  to="/camera/rgb/camera_info"/>
		<remap from="~predictions"   to="/ros_caffe/predictions"/>
	</node>

	<!-- RVIZ to view the visualization -->
	<node name="RVIZ" pkg="rviz" type="rviz" args=" -d $(find ros_caffe)/rviz/ros_caffe_openni.rviz"/>

</launch>
